BridgeCareApp/BridgeCare/DataAccessLayer/InvestmentStrategies.cs

```diff
@@ -106,5 +98,36 @@ public void SetInvestmentStrategies(InvestmentStrategyModel data, BridgeCareCont
             }
             return;
         }
+        public void UpsertYearlyData(InvestmentStrategyModel investment,
+           SIMULATION simulation, BridgeCareContext db)
+        {
+            int dataIndex = 0;
+
+            foreach (YEARLYINVESTMENT yearly in simulation.YEARLYINVESTMENTs.ToList())
+            {
+                if (investment.YearlyBudgets.Count() > dataIndex)
+                {
+                    yearly.YEAR_ = investment.YearlyBudgets[dataIndex].Year;
+                    yearly.BUDGETNAME= investment.YearlyBudgets[dataIndex].BudgetName;
+                    yearly.AMOUNT = investment.YearlyBudgets[dataIndex].BudgetAmount;
+                }
+                else
+                {
+                    db.Entry(yearly).State = EntityState.Deleted;
+                }
+                dataIndex++;
+            }
+            //these must be inserts as the updated records exceed existing records
+            while (investment.YearlyBudgets.Count() > dataIndex)
+            {
+                var yearly = new YEARLYINVESTMENT(investment.SimulationId,
```

Comment:
yearly can be renamed as yearlyInvestment.

Reply:
ok